sled-storage/src/manager.rs

```diff
@@ -914,7 +915,11 @@ impl StorageManager {
         let result = self.datasets_ensure_internal(&log, &config).await;
 
         let ledger_data = ledger.data_mut();
-        if *ledger_data == config {
+        if had_old_ledger && *ledger_data == config {
```

Comment:
This change is only tangentially related to this PR, but it's necessary for some tests.

Because we're reading the ledger of datasets when we provision zones, we were failing in tests that **had not created a ledger of datasets**.

To mitigate: when we're creating the "empty dataset ledger", we want to still write it to our simulated M.2s.

However, in the old version of this code, as an optimization, we would skip the re-write to the M.2 if nothing had changed. Unfortunately, if there was no prior ledger, we would initialize the ledger data with a "Default value", meaning that it wasn't actually possible to write the "default value" of the ledger to storage.

(This hadn't been an issue before, because our code typically **has** datasets it cares about inserting when this method gets called).

Anyway, to mitigate, I'm now skipping the ledger re-write if and only if: (1) There **was** an old ledger, and (2) it hasn't changed

Reply:
Agreed this is cleaner, patched to this version.